File: java/src/test/java/org/fluro/CheckoutTests.java
package org.fluro;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.HashSet;

class CheckoutTests {
    private Checkout checkout = new Checkout(
            new Stock(
                    new HashSet<>(Arrays.asList(new Item('A', 50))
                    )
            ));

    @Test
    public void canCheckoutSingleItem() {
        checkout.scanItem('A');

        assert (checkout.total() == 50);
    }

    @Test
    public void canScanMultipleItems() {
        checkout.scanItem('A')
                .scanItem('A');

        assert (checkout.total() == 100);
    }

    @Test
    public void totalThrowsAnErrorIfSkuIsNotInStock() {
        checkout.scanItem('Z');
        Assertions.assertThrows(RuntimeException.class, () ->
                checkout.total()
        );
    }
}

File: kotlin/src/test/kotlin/org/fluro/CheckoutTests.kt
package org.fluro

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows

class CheckoutTests {
    private val checkout = Checkout(
        Stock(
            setOf(Item('A', 50))
        )
    )

    @Test
    fun `can checkout single item`() {
        checkout.scanItem('A')

        assert(checkout.total() == 50)
    }

    @Test
    fun `can scan multiple items`() {
        checkout
            .scanItem('A')
            .scanItem('A')

        assert(checkout.total() == 100)
    }

    @Test
    fun `total throws an error if sku is not in stock`() {
        checkout.scanItem('Z')
        assertThrows<RuntimeException> {
            checkout.total()
        }
    }
}
